test/new-e2e/tests/agent-metrics-logs/log-agent/windows-log/file-tailing/file_tailing_test.go

```diff
@@ -137,17 +138,15 @@ func (s *WindowsFakeintakeSuite) testLogNoPermission() {
 	assert.NoErrorf(t, err, "Unable to adjust permissions for the log file %s.", logFilePath)
 	t.Logf("Read permissions revoked")
 
-	// Generate logs and check the intake for no new logs because of revoked permissions
```

Comment:
Notes for reviewers:

Given https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/596176036
The panic's stack is:
```
panic: Fail in goroutine after TestE2EVMFakeintakeSuite/TestWindowsLogTailing/LogCollectionNoPermission has completed
goroutine 509 [running]:
testing.(*common).Fail(0xc000d54000)
	/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.linux-amd64/src/testing/testing.go:952 +0xd4
testing.(*common).Errorf(0xc000d54000, {0x500e04e?, 0x414ba5?}, {0xc0008961c0?, 0x3dd0ba0?, 0x81ae01?})
	/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.linux-amd64/src/testing/testing.go:1075 +0x5e
github.com/stretchr/testify/assert.EventuallyWithT({0x5560820, 0xc000d54000}, 0xc0009e8080, 0x1bf08eb000, 0x2540be400, {0x0, 0x0, 0x0})
	/go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:1973 +0x303
github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-metrics-logs/log-agent/utils.CheckLogsNotExpected(0xc000d54000, 0xc000b7b3b0, {0x500f6fd, 0x5}, {0x501d03e, 0xd})
	/go/src/github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-metrics-logs/log-agent/utils/file_tailing_utils.go:175 +0xe5
github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-metrics-logs/log-agent/windows-log/file-tailing.(*WindowsFakeintakeSuite).testLogNoPermission.func1(0xc000aee408)
	/go/src/github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-metrics-logs/log-agent/windows-log/file-tailing/file_tailing_test.go:147 +0xd0
github.com/stretchr/testify/assert.EventuallyWithT.func1()
	/go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:1983 +0x7f
created by github.com/stretchr/testify/assert.EventuallyWithT in goroutine 504
	/go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:1978 +0x277
```

From this I assume the panic happens in one of the `goroutine` created by `Eventually` at [https://github.com/DataDog/datadog-agent/blob/dee60a789471183834655fa2890a948b69f5[…]/tests/agent-metrics-logs/log-agent/utils/file_tailing_utils.go](https://github.com/DataDog/datadog-agent/blob/dee60a789471183834655fa2890a948b69f5efdd/test/new-e2e/tests/agent-metrics-logs/log-agent/utils/file_tailing_utils.go#L175)

`Eventually` creates a new `goroutine` and execute the function passed as second argument in that context. Here we hare a nested `Eventually` from `testLogNoPermission()` that calls `CheckLogsNotExpected` passing the parent context instead of the collector context
[https://github.com/DataDog/datadog-agent/blob/dee60a789471183834655fa2890a948b69f5[…]cs-logs/log-agent/windows-log/file-tailing/file_tailing_test.go](https://github.com/DataDog/datadog-agent/blob/dee60a789471183834655fa2890a948b69f5efdd/test/new-e2e/tests/agent-metrics-logs/log-agent/windows-log/file-tailing/file_tailing_test.go#L147)

This makes the parent `Eventually` in `windows-log/file-tailing/file_tailing_test.go` return as soon as its collected assertion are good.